backend/src/event/Event.ts

```diff
@@ -1,212 +1,48 @@
-import { EventProtocol as DbEvent } from '@entity/EventProtocol'
+import { Event as DbEvent } from '@entity/Event'
+import { User as DbUser } from '@entity/User'
+import { ContributionMessage as DbContributionMessage } from '@entity/ContributionMessage'
+import { Contribution as DbContribution } from '@entity/Contribution'
+import { Transaction as DbTransaction } from '@entity/Transaction'
 import Decimal from 'decimal.js-light'
-import { EventProtocolType } from './EventProtocolType'
+import { EventType } from './Event'
 
 export const Event = (
-  type: EventProtocolType,
-  userId: number,
-  xUserId: number | null = null,
-  xCommunityId: number | null = null,
-  transactionId: number | null = null,
-  contributionId: number | null = null,
+  type: EventType,
+  affectedUser: DbUser,
+  actingUser: DbUser,
+  involvedUser: DbUser | null = null,
```

Comment:
keep in mind, that all three users could be a remote user in context of a x-community-tx.